### PR TITLE
Added levenshtein to setjob (plasma's idea)

### PR DESCRIPTION
**Pull request by ggggg**
I didnt think this needed another branch as I have already deleted the setjob one

### CODE REVIEW

File: src/BPEssentials/Commands/Player/Inventory/Give.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public void Invoke(ShPlayer player, string name, int amount = 1, ShPlayer target
                     return;
                 }
                 item = Core.Instance.EntityHandler.Items.OrderByDescending(x => LevenshteinDistance.CalculateSimilarity(x.Key, name)).FirstOrDefault();
-
```
Comment (ggggg) [inline src/BPEssentials/Commands/Player/Inventory/Give.cs]: Wops, should have not committed that lol

File: src/BPEssentials/Commands/Player/Inventory/Give.cs [hunk 2]
```diff
@@ -22,7 +22,7 @@ public void Invoke(ShPlayer player, string name, int amount = 1, ShPlayer target
...
+
```
Comment (PLASMAchicken) [inline src/BPEssentials/Commands/Player/Inventory/Give.cs]:
```suggestion

```

File: src/BPEssentials/Commands/Player/SetJob.cs [hunk 3]
```diff
@@ -3,20 +3,49 @@
...
             
             target = target ?? player;
-            target.svPlayer.SvTrySetJob(job, true, false);
-            player.TS("job_set", target.username.CleanerMessage(), job.ToString());
+            Dictionary<string, JobIndex> jobs = new Dictionary<string, JobIndex>();
+            foreach (JobIndex jobType in Enum.GetValues(typeof(JobIndex)))
+            {
+                jobs.Add(jobType.ToString(), jobType);
+                Console.WriteLine($"{ jobType.ToString()} , { jobType}");
+            }
```
Comment (PLASMAchicken) [inline src/BPEssentials/Commands/Player/SetJob.cs]: Preferably we should move this somewhere else

File: src/BPEssentials/Commands/Player/SetJob.cs [hunk 4]
```diff
@@ -3,20 +3,49 @@
...
         {
             
             target = target ?? player;
-            target.svPlayer.SvTrySetJob(job, true, false);
-            player.TS("job_set", target.username.CleanerMessage(), job.ToString());
+            Dictionary<string, JobIndex> jobs = new Dictionary<string, JobIndex>();
+            foreach (JobIndex jobType in Enum.GetValues(typeof(JobIndex)))
+            {
+                jobs.Add(jobType.ToString(), jobType);
+                Console.WriteLine($"{ jobType.ToString()} , { jobType}");
```
Comment (PLASMAchicken) [inline src/BPEssentials/Commands/Player/SetJob.cs]:
```suggestion
```
Remove Log's for Debugging

File: src/BPEssentials/Commands/Player/SetJob.cs [hunk 5]
```diff
@@ -3,20 +3,49 @@
...
+            }           
+
+            var wantedJob = jobs.FirstOrDefault(x => x.Key == jobName);
+            Console.WriteLine(wantedJob.Value);
```
Comment (PLASMAchicken) [inline src/BPEssentials/Commands/Player/SetJob.cs]:
```suggestion
```
Remove Log's for Debugging

File: src/BPEssentials/Commands/Player/SetJob.cs [hunk 6]
```diff
@@ -3,20 +3,49 @@
...
+                    player.TS("levenshteinSuggest", wantedJob.Key);
+                    return;
+                }
+            }
+            target.svPlayer.SvTrySetJob(wantedJob.Value, true, false);
+            player.TS("job_set", target.username.CleanerMessage(), jobName.ToString());
             if (target != player) 
             {
-                target.TS("new_job", player.username.CleanerMessage(), job.ToString());
+                target.TS("new_job", player.username.CleanerMessage(), jobName.ToString());
```
Comment (PLASMAchicken) [inline src/BPEssentials/Commands/Player/SetJob.cs]:
```suggestion
                target.TS("new_job", player.username.CleanerMessage(), wantedJob.Key);
```

File: src/BPEssentials/Commands/Player/Inventory/Give.cs (hunk 2 above)
Comment (ggggg) [reply src/BPEssentials/Commands/Player/Inventory/Give.cs]: done

File: src/BPEssentials/Commands/Player/SetJob.cs (hunk 3 above)
Comment (ggggg) [reply src/BPEssentials/Commands/Player/SetJob.cs]: Like where?

File: src/BPEssentials/BPEssentials.csproj [hunk 7]
```diff
@@ -299,7 +299,6 @@
     <Compile Include="Commands\Moderation\Ban.cs" />
     <Compile Include="Commands\Moderation\Kick.cs" />
     <Compile Include="Commands\Teleport\TeleportCoordinates.cs" />
-    <Compile Include="Commands\Player\ToggleGodmode.cs" />
```
Comment (PLASMAchicken) [inline src/BPEssentials/BPEssentials.csproj]: We can also remove GodMode from ExtendedPlayer

File: src/BPEssentials/Commands/Player/SetJob.cs [hunk 8]
```diff
@@ -3,20 +3,47 @@
...
     public class SetJob : Command
     {
-        public void Invoke(ShPlayer player, JobIndex job, ShPlayer target = null)
+        public void Invoke(ShPlayer player, string jobName, ShPlayer target = null)
         {
             
             target = target ?? player;
-            target.svPlayer.SvTrySetJob(job, true, false);
-            player.TS("job_set", target.username.CleanerMessage(), job.ToString());
+            Dictionary<string, JobIndex> jobs = new Dictionary<string, JobIndex>();
```
Comment (UserR00T) [inline src/BPEssentials/Commands/Player/SetJob.cs:19]: Use `var` wherever possible.
```suggestion
            var jobs = new Dictionary<string, JobIndex>();
```

File: src/BPEssentials/Commands/Player/SetJob.cs [hunk 9]
```diff
@@ -3,20 +3,47 @@
...
+            foreach (JobIndex jobType in Enum.GetValues(typeof(JobIndex)))
```
Comment (UserR00T) [inline src/BPEssentials/Commands/Player/SetJob.cs:20]: Use `var` wherever possible.
```suggestion
            foreach (var jobType in Enum.GetValues(typeof(JobIndex)))
```

**Pull request merged**